src/drive/web/modules/services/components/SuggestionProvider.jsx

```diff
@@ -38,52 +56,50 @@ class SuggestionProvider extends React.Component {
           title: result.name,
           subtitle: result.path,
           term: result.name,
-          onSelect: 'open:' + result.url,
+          onSelect: result.onSelect || 'open:' + result.url,
           icon: result.icon
         }))
       },
       intent.attributes.client
     )
   }
 
-  // fetches pretty much all the files and preloads FuzzyPathSearch
+  /**
+   * Fetches all files without trashed and preloads FuzzyPathSearch
+   *
+   * Using _find route (from findAll) improves performance:
+   * - using partial index to reduce amount of data fetched
+   * - removing trashed data directly
+   *
+   * Also, this method:
+   * - set first the `hasIndexFilesBeenLaunched` to prevent multiple calls
+   * - removes orphan file
+   * - normalize file to match <SearchBar> expectation
+   * - preloads FuzzyPathSearch
+   *
+   * @returns {Promise<void>} nothing
+   */
   async indexFiles() {
     const { client } = this.props
-    // TODO: fix me
-    // eslint-disable-next-line no-async-promise-executor
-    return new Promise(async resolve => {
-      const resp = await cozy.client.fetchJSON(
-        'GET',
-        `/data/io.cozy.files/_all_docs?include_docs=true`
-      )
-      const files = resp.rows
-        // TODO: fix me
-        // eslint-disable-next-line no-prototype-builtins
-        .filter(row => !row.doc.hasOwnProperty('views'))
-        .map(row => ({ id: row.id, ...row.doc }))
+    this.hasIndexFilesBeenLaunched = true
 
-      const folders = files.filter(file => file.type === TYPE_DIRECTORY)
+    const { selector, options } = prepareSuggestionQuery()
+    const files = await client
+      .collection(DOCTYPE_FILES)
+      .findAll(selector, options)
 
-      const notInTrash = file =>
-        !file.trashed && !/^\/\.cozy_trash/.test(file.path)
-      const notOrphans = file =>
-        folders.find(folder => folder._id === file.dir_id) !== undefined
+    const folders = files.filter(file => file.type === TYPE_DIRECTORY)
```

Comment:
Since we can benefits from http2 multiplexing, I'm wondering if we should not make two requests: 
- one for the files 
- one for the folders 

We'll have the almost the same number of round trips (1 or 2) but we'll have a few of them parallelized. @paultranvan what do you think?

Reply:
I assume that, *most of the time*, the number of files will be order of magnitudes higher than the number of folders. So, the folder query might end up sooner than the files one, which is good, as we want to display the folders first. The downside is that we have now 2 indexes to compute instead of one, which might slow down the search when we make a query with non-indexed docs. 
I think what you propose is still relevant either way, as it should allow to get partial (i.e. folders) results faster and I think faster user response is more important than the overall performance. 
That being said, I also think paginated mango queries is sub-optimal for this use-case :/